AlphaWallet/Tokens/Types/TokenObject.swift

```diff
@@ -47,7 +47,10 @@ class TokenObject: Object {
     }
 
     var address: Address {
-        return Address(string: contract)!
+        if let ethAddress = Address(string: contract) {
+            return ethAddress
+        }
+        return EtherKeystore.current!.address
```

Comment:
This part seems a bit strange. It's returning a wallet address as a fallback when the contract address is invalid?